src/algebra/polynomial/big_operators.lean

```diff
@@ -34,18 +35,39 @@ variables {R : Type u} {ι : Type w}
 
 namespace polynomial
 
-variable (s : finset ι)
+variables (s : finset ι) (t : multiset ι)
 
 section comm_semiring
 variables [comm_semiring R] (f : ι → polynomial R)
 
+lemma nat_degree_multiset_prod_le :
+  (t.map f).prod.nat_degree ≤ (t.map (λ i, (f i).nat_degree)).sum :=
+begin
+  refine multiset.induction_on t _ (λ a t ih, _), { simp },
+  rw [map_cons, prod_cons, map_cons, sum_cons],
+  transitivity (f a).nat_degree + (t.map f).prod.nat_degree,
+  { apply polynomial.nat_degree_mul_le },
+  { exact add_le_add (le_refl _) ih }
+end
```

Comment:
Should we generalize to `list.prod` too?

Reply:
I considered it, but I do not think it is worth the effort (for my usecases at least): using a multiset in the context of polynomials is relatively common (e.g. `polynomial.roots`), and a list not so much, and in the commutative case, you don't gain much by generalizing to lists, compared to the possibility of duplicate terms in a multiset.

If you do have a usecase, I'm happy to generalize to lists too.